### PR TITLE
Multi outputs indicators

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fulfills a goal of #351 and goes in the direction of #342 and #516.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

## What kind of change does this PR introduce?
Copy of #510  at midpoint, I created a new PR to limit the number of revert commits. 

So, modifies the class Indicator so that it accepts any number of outputs. Variable metadata can still be passed as strings, but also as lists of strings. The number of outputs is assumed to be `len(indicator.var_name)` . When a string is passed to a multi-output indicator, it is copied on all outputs.

In opposition to the other PR, this version allows to still use classic class inheritance for indicators. Internally, the indicator class uses `Ind.cf_attrs` to manage the number of outputs and their attributes. It is created in the `__new__` method.  Note that this means that variable metadata is _read-only_, which it wasn't before.

The implementation needed many small tweaks elsewhere to work:

- Indicator's method `update_attrs` and `format` are now _classmethods_, so one must pass the set of attributes to update when using them.
- `xclim.core.units.declare_units` now works with multiple outputs : if the output of the indice is a tuple, it iterates over a zip of (out_units, output), so the units must be declared as a sequence.
- `xclim.core.locales.get_local_attrs` now uses the indicator's class name instead of the indicator object itself. Also, the `fill_missing` keyword has been removed. The "fill_missing" behavior is now part of the `Indicator.translate_attrs` method.
- Moved some arguments of `xclim.indices.fire_weather_indexes` from the starred kwargs to the explicit kwargs.
- Added the `"[speed]" dimensionnality.

## Does this PR introduce a breaking change?
The declaration, use and output of current single-output indicator hasn't changed. But:

- Accessing variable attribute on the indicator object might return a single string or a list of strings. The attributes concerned are the ones in `Ind._cf_names`.
- `Ind.cf_attrs` is now a list of dictionaries, very often a single dict.
- `Ind.json()` now as an entry "outputs" which is a list of dicts (in fact, a (formatted) copy of `cf_attrs`).

## Other information
The `long_name` json attribute is replaced by `title`. Currently, titles are not a reliable and coherent source of information for indicators, but #516 will take care of that.

### Changes example for `xc.atmos.tg_mean.json()`
<details>

`.to_json()` before:
```
{'identifier': 'tg_mean',
 'var_name': 'tg_mean',
 'abstract': '',
 'keywords': '',
 'standard_name': 'air_temperature',
 'long_name': 'Mean daily mean temperature',
 'units': 'K',
 'cell_methods': 'time: mean within days time: mean over days',
 'description': '{freq} mean of daily mean temperature.',
 'notes': '',
 'parameters': "{'tas': {'default': None, 'desc': ''}, 'freq': {'default': 'YS', 'desc': ''}}"}
```
`to_json()` in this PR:
```
{'identifier': 'tg_mean',
 'title': '',
 'abstract': '',
 'keywords': '',
 'outputs': [{'var_name': 'tg_mean',
   'standard_name': 'air_temperature',
   'long_name': 'Mean daily mean temperature',
   'units': 'K',
   'cell_methods': 'time: mean within days time: mean over days',
   'description': '{freq} mean of daily mean temperature.'}],
 'notes': '',
 'parameters': "{'tas': {'default': None, 'desc': ''}, 'freq': {'default': 'YS', 'desc': ''}}"}
```
</details>

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -31,7 +35,9 @@ Internal changes
 * Indicator subclasses `TasminTasmax` and `PrTas` now inherit from `Daily2D`.
 * Docstring style now enforced using the `pydocstyle` with `numpy` doctsring conventions.
 * Doctests are now performed for all docstring `Examples` using `xdoctest`. Failing examples must be explicitly skipped otherwise build will now fail.
+* Indicator methods `update_attrs` and `format` are now classmethods, attrs to update must as passed.
```
Comment (huard) [inline HISTORY.rst:38]: typo

File: xclim/core/indicator.py [hunk 2]
```diff
@@ -95,112 +98,144 @@ class Indicator:
...
+            values = kwds.get(name, getattr(cls, name))
+            if not isinstance(values, (list, tuple)):
+                values = [values] * n_outs
+            elif len(values) != n_outs:
+                raise ValueError(
+                    f"Attribute {name} has {len(values)} elements but should have {n_outs} according to passed var_name."
+                )
+            for attrs, value in zip(kwds["cf_attrs"], values):
+                if value:
+                    attrs[name] = value
```
Comment (huard) [inline xclim/core/indicator.py:235]: Does it make sense to put this logic into a method ?

File: xclim/core/indicator.py [hunk 3]
```diff
@@ -261,26 +294,49 @@ def __call__(self, *args, **kwds):
...
 
         # Compute the indicator values, ignoring NaNs and missing values.
-        out = self.compute(**das, **ba.kwargs)
+        outs = self.compute(**das, **ba.kwargs)
+        if isinstance(outs, tuple) and len(outs) != n_outs:
+            raise ValueError(
+                f"Indicator {self.identifier} was wrongly defined. Expected {n_outs} outputs, got {len(outs)}."
+            )
+        if n_outs == 1:
+            outs = [outs]
```
Comment (huard) [inline xclim/core/indicator.py:317]:
```suggestion
        if n_outs == 1:
            outs = [outs]
        if len(outs) != n_outs:
            raise ValueError(
                f"Indicator {self.identifier} was wrongly defined. Expected {n_outs} outputs, got {len(outs)}."
            )

```

Comment (huard) [reply xclim/core/indicator.py:317]: or maybe better would be 
```
if isinstance(outs, xr.DataArray):
    outs = (outs, )
```

File: xclim/core/indicator.py [hunk 4]
```diff
@@ -261,26 +294,49 @@ def __call__(self, *args, **kwds):
...
         # Mask results that do not meet criteria defined by the `missing` method.
+        # This means all variables must have the same dimensions...
         mask = self.mask(*das.values(), **ba.arguments)
+        outs = [out.where(~mask) for out in outs]
 
-        return out.where(~mask).rename(vname)
+        # Return a single DataArray in case of single output, otherwise a tuple
+        if n_outs == 1:
+            return outs[0]
+        return tuple(outs)
```
Comment (huard) [inline xclim/core/indicator.py:339]: Shouldn't `outs` already be a tuple ?

File: xclim/core/indicator.py [hunk 5]
```diff
@@ -95,112 +98,144 @@ class Indicator:
...
     abstract = ""
     keywords = ""
     references = ""
-    comment = ""
     notes = ""
 
     def __new__(cls, **kwds):
-        """Create subclass from arguments."""
-        identifier = kwds.get("identifier", getattr(cls, "identifier"))
+        """Create subclass from arguments. Keywods of kwds that are class attributes are attributed here."""
```
Comment (aulemahal) [inline xclim/core/indicator.py:197]:
```suggestion
        """Create subclass from arguments."""
```

File: xclim/core/indicator.py (hunk 2 above)
Comment (aulemahal) [reply xclim/core/indicator.py:235]: Could. And it would reduce the complexity of the method...

File: xclim/core/indicator.py (hunk 4 above)
Comment (aulemahal) [reply xclim/core/indicator.py:339]: Because I use list-comprehension, it's a list. But multiple outputs in python is conventionally a tuple.

**Pull request merged**